gc/gc-base/src/main/java/org/projectnessie/gc/base/GCImpl.java

```diff
@@ -95,17 +81,11 @@ public GCImpl(GCParams gcParams) {
    * @param session spark session for distributed computation
    * @return {@link IdentifiedResult} object having expired contents per content id.
    */
-  public IdentifiedResult identifyExpiredContents(SparkSession session) {
+  public static IdentifiedResult identifyExpiredContents(SparkSession session, GCParams gcParams) {
```

Comment:
I would probably keep refactoring in a separate PR.

Also as this flow is WIP, may be better to wait till we merge #3421

Reply:
Those changes are needed for serialization changes, so it cannot be split.